app/models/mixins/virtual_total_mixin.rb

```diff
@@ -16,7 +16,7 @@ module ClassMethods
     #
     def virtual_total(name, relation, options = {})
       define_method(name) do
-        send(relation).try(:size) || 0
+        (attribute_present?(name) ? self[name] : nil) || send(relation).try(:size) || 0
```

Comment:
Thanks for pointer.
It was calling `attributes` and was duping. Thanks to your advice I moved over to `[]` to avoid that duping.